src/api/routes/groups.route.ts

```diff
@@ -0,0 +1,29 @@
+import { Router } from "express";
+
+import {
+  validateCreateGroup,
+  validateUpdateGroup,
+  validateAddUsersToGroup,
+  validateQuery,
+  validateParams,
+} from "../../validations";
+import { groupController } from "../controllers";
+
+const router = Router();
+
+router
+  .route("/")
+  .get(validateQuery, groupController.getGroups)
+  .post(validateCreateGroup, groupController.createGroup);
+
+router
+  .route("/add-users-to-group")
```

Comment:
While it's acceptable in the context of the current program, keep in mind that this API design does not strictly complies with RESTful API design patterns.
For more info: https://www.mscharhag.com/api-design/rest-many-to-many-relations

Reply:
Thanks, very insightful article (I can see why this approach would be an improvement to the pattern used here). 

I was a little unsure of how to approach this with it being a transaction, but I've change the route here to: 

`/groups/<groupId>/users/add-users-to-group` 

What do you think? I was a little unsure about having a post request with a body and params (Some casual reading makes it sound as though this is okay though).

Reply:
Yeah, this feels a bit more inline with the RESTful guidelines.
It's ok to have `POST /items/:itemId/things/<WORD>` along with `POST /items/:itemId/things/:thingId` for cases when a more generic (`/:thingId`) POST is already present. Just make sure the `things/<WORD>` does not fall into `things/:thingId` case (when internally your code assumes `thingId = <WORD>`).